nexus/src/db/datastore.rs

```diff
@@ -2679,6 +2679,70 @@ impl DataStore {
             }),
         }
     }
+
+    // SSH public keys
+
+    pub async fn ssh_keys_list(
+        &self,
+        silo_user_id: Uuid,
+        page_params: &DataPageParams<'_, Name>,
+    ) -> ListResultVec<SshKey> {
+        use db::schema::ssh_key::dsl;
+
+        paginated(dsl::ssh_key, dsl::name, page_params)
+            .filter(dsl::silo_user_id.eq(silo_user_id))
+            .filter(dsl::time_deleted.is_null())
+            .select(SshKey::as_select())
+            .load_async(self.pool())
+            .await
+            .map_err(|e| public_error_from_diesel_pool(e, ErrorHandler::Server))
+    }
+
+    /// Create a new SSH public key for a user.
+    pub async fn ssh_key_create(
+        &self,
+        ssh_key: SshKey,
+    ) -> CreateResult<SshKey> {
+        use db::schema::ssh_key::dsl;
+
```

Comment:
Similarly, I think you want to accept `authz::SiloUser` here and do an authz check for `CreateChild` on the `SiloUser`.  You may also want to assert that the silo user id matches the ssh key id.  I think that's what we do in most other create-X-inside-Y functions, like `vpc_create_subnet` as an example.

Reply:
Yep, fixed in 95114e05.